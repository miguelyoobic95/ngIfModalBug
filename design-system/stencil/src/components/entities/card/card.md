---
name: Card
category: Entities Components
---

## Attributes
|Name|Type|Default|Description|
|---|---|---|---|
|`heading`|string|   |heading of the card|
|`subheadings`|string[ ]|`[ ]`|subheadings of the card|
|`imgSrc`|string|   |source of the image of the card|
|`top-left-badge`|string| |sets the content for the top left badge, will display nothing if undefined|
|`top-right-badge`|string| |sets the content for the top right badge, will display nothing if undefined|
|`bottom-left-badge`|string| |sets the content for the bottom left badge, will display nothing if undefined|
|`bottom-right-badge`|string| |sets the content for the bottom right badge, will display nothing if undefined|
|`avatarImgs`|string[ ]|`[ ]`|source of the avatar image displayed in the card|
|`isActivable`|boolean|`false`|display a checkbox at the top and allow the card to be activated|
|`isUserCard`|boolean|`false`|displays an avatar instead of an image with the specified `imgSrc`|
|`date`|string||date displayed on the top right of the card|
|`tags`|IBadgeEntry[]||Tags displays after the content of the card|
|`avatarShape`|string|`'rectangle'`|You can set this to `'circle'`|
|`actionButtonTitle`|string||Tags displays after the content of the card|
|`animationName`|string||Displays a custom animation on load, see animation docs for all options|
|`hasMenu`|boolean|`false`|Displays a context menu on the card, with slot name `menu-slot`|
|`type`|string: `default`, `list`| `default`| sets the type of the card|

### Basic Card Example

```yoo-card-heading.html
    <yoo-card heading="My Heading">
    </yoo-card>
    <br>
    <yoo-card heading="My Heading" class="horizontal">
    </yoo-card>
```

### With Custom animation

```yoo-card-animation.html
    <yoo-card heading="Animation: flip_3d_vertical" animation-name="flip_3d_vertical">
    </yoo-card>
    <br>
    <yoo-card heading="Animation: super_scaled" animation-name="super_scaled" class="horizontal">
    </yoo-card>
```

### List Type
```yoo-card-list.html
    <yoo-card title="List Mode" type="list" sticker="Yoo" avatar-shape="circle" img-src="http://www.twitrcovers.com/wp-content/uploads/2012/10/Star-Wars-l.jpg" id="list">
    </yoo-card>
```
```yoo-card-list.js hidden
    var listMode = document.querySelector('#list');
    var tags = [{text: 'Finished', icon:'yo-flag'},{text: 'Fire Account', icon:'yo-fire'}];
    listMode.tags = tags;
    listMode.subtitles = ["subtitle 1", "subtitle 2"];
```

### With Custom Image

```yoo-card-image.html
    <yoo-card heading="My heading" img-src="http://www.twitrcovers.com/wp-content/uploads/2012/10/Star-Wars-l.jpg">
    </yoo-card>

     <yoo-card heading="My heading" class="horizontal" img-src="https://images-na.ssl-images-amazon.com/images/I/51q8Jb98R4L.jpg">
    </yoo-card>
```


### Badge

```yoo-card-badge.html
    <yoo-card class="accent" top-left-badge="TL" top-right-badge="TR" bottom-left-badge="BL" bottom-right-badge="BR"></yoo-card>

    <yoo-card class="accent horizontal" top-left-badge="TL" top-right-badge="TR" bottom-left-badge="BL" bottom-right-badge="BR"></yoo-card>
```

### Date

```yoo-card-date.html
    <yoo-card class="accent" heading="My heading" date="25/03/1995"></yoo-card>

    <yoo-card class="accent horizontal" heading="My heading" date="25/03/1995"></yoo-card>
```

### Tags

```yoo-card-tags.html
    <yoo-card class="accent" heading="My heading"></yoo-card>

    <yoo-card class="accent horizontal" heading="My heading"></yoo-card>
```

```yoo-card-tags.js
    var comps = [...document.querySelectorAll('yoo-card')];
    var tags = [{text: 'Finished', icon:'yo-flag'},{text: 'Fire Account', icon:'yo-fire'}];
    comps.map((comp) => {comp.tags = tags});
```


### With slotted content

```yoo-card-slotted.html
    <yoo-card>
        <div slot="content-slot">
            I am the slotted content
            <div attr-layout="row">
                <yoo-badge text="789" class="accent"></yoo-badge>
                <yoo-badge text="897" class="danger"></yoo-badge>
                <yoo-badge text="978" class="gradient-info"></yoo-badge>
            </div>
        </div>
    </yoo-card>

    <yoo-card class="horizontal">
        <div slot="content-slot">
            I am the slotted content
            <div attr-layout="row">
                <yoo-badge text="789" class="accent"></yoo-badge>
                <yoo-badge text="897" class="danger"></yoo-badge>
                <yoo-badge text="978" class="gradient-info"></yoo-badge>
            </div>
        </div>
    </yoo-card>
```

```yoo-card-slotted.css hidden
    yoo-badge{
        margin-left: 0.15rem;
        margin-right: 0.15rem;
    }
```

### User Card

```yoo-user-card.html
    <yoo-card heading="My heading" avatar-shape="circle" img-src="https://steamcdn-a.akamaihd.net/steamcommunity/public/images/avatars/84/84210f8488cf340005271ab03092659fd3e972f0_full.jpg">
    </yoo-card>

    <yoo-card heading="My heading" class="horizontal" avatar-shape="circle" img-src="https://steamcdn-a.akamaihd.net/steamcommunity/public/images/avatars/84/84210f8488cf340005271ab03092659fd3e972f0_full.jpg">
    </yoo-card>
```

### With Bottom Avatar

```yoo-card-avatar.html
    <yoo-card heading="With Avatars">
        <div slot="content-slot">
            I am the slotted content
            <div attr-layout="row">
                <yoo-badge text="789" class="accent"></yoo-badge>
                <yoo-badge text="897" class="danger"></yoo-badge>
                <yoo-badge text="978" class="gradient-info"></yoo-badge>
            </div>
        </div>
    </yoo-card>
```
```yoo-card-avatar.js hidden
    var comp = document.querySelector('yoo-card');
    comp.avatarImgs = ["./assets/grid/empty-state-1.svg", "./assets/grid/empty-state-1.svg", "./assets/grid/empty-state-1.svg"];
```

### With Bottom Avatar Horizontal

```yoo-card-avatar-horizontal.html
    <yoo-card class="horizontal" heading="With Avatars">
        <div slot="content-slot">
            I am the slotted content
            <div attr-layout="row">
                <yoo-badge text="789" class="accent"></yoo-badge>
                <yoo-badge text="897" class="danger"></yoo-badge>
                <yoo-badge text="978" class="gradient-info"></yoo-badge>
            </div>
        </div>
    </yoo-card>
```

```yoo-card-avatar-horizontal.js
    var comp = document.querySelector('yoo-card');
    comp.avatarImgs = ["./assets/grid/empty-state-1.svg", "./assets/grid/empty-state-1.svg", "./assets/grid/empty-state-1.svg"];
```
### Activable

```yoo-card-activable.html
    <yoo-card class="accent" heading="Activable Card" is-activable="true">
    </yoo-card>
    <br>
    <yoo-card class="accent" class="horizontal" heading="Activable Card" is-activable="true">
    </yoo-card>
```

### Full Card example

```yoo-card-full-card.html


     <yoo-card class="danger hoverable" heading="My First Card" top-left-badge="TL" top-right-badge="TR" bottom-left-badge="BL" bottom-right-badge="BR" action-button-title="action button"  is-activable="true" sticker="Yoo" has-menu="true" >
        <div slot="menu-slot">This is a full card</div>
        <div slot="content-slot">
            I am the slotted content and doesn't this card look great?
            <div attr-layout="row">
                <yoo-badge text="789" class="accent"></yoo-badge>
                <yoo-badge text="897" class="danger"></yoo-badge>
                <yoo-badge text="978" class="gradient-info"></yoo-badge>
            </div>
        </div>
    </yoo-card>
```

```yoo-card-full-card.css hidden
    yoo-badge {
        margin-left: 0.15rem;
        margin-right: 0.15rem;
    }
```

```yoo-card-full-card.js
    var comp = document.querySelector('yoo-card');
    comp.avatarImgs = ["./assets/grid/empty-state-1.svg", "./assets/grid/empty-state-1.svg", "./assets/grid/empty-state-1.svg"];
    comp.subheadings = ["subtitle 1", "subtitle 2"];
```

### Full Card Horizontal example

```yoo-card-full-card-horizontal.html


     <yoo-card class="danger hoverable horizontal" heading="My First Card" top-left-badge="TL" top-right-badge="TR" bottom-left-badge="BL" bottom-right-badge="BR" action-button-title="action button"  is-activable="true" has-menu="true"  >
        <div slot="menu-slot">This is a full card</div>
        <div slot="content-slot">
            I am the slotted content
            <div attr-layout="row">
                <yoo-badge text="789" class="accent"></yoo-badge>
                <yoo-badge text="897" class="danger"></yoo-badge>
                <yoo-badge text="978" class="gradient-info"></yoo-badge>
            </div>
            i love content
            <br>
            i love content
            <br>
            i love content
            <br>
            i love content
        </div>
    </yoo-card>
```

```yoo-card-full-card-horizontal.css hidden
    yoo-badge {
        margin-left: 0.15rem;
        margin-right: 0.15rem;
    }
```

```yoo-card-full-card-horizontal.js
    var comp = document.querySelector('yoo-card');
    comp.avatarImgs = ["./assets/grid/empty-state-1.svg", "./assets/grid/empty-state-1.svg", "./assets/grid/empty-state-1.svg"];
    comp.subheadings = ["subtitle 1", "subtitle 2"];
```

## CSS

|Type|Name|Description|
|---|---|---|
|Hoverable|`hoverable`|yoo-card with added hover effect|
|No Status|`no-status`|yoo-card with no status bar|
|Horizontal|`horizontal`|yoo-card displayed in horizontal|
|Accent|`accent`|yoo-card accent|
|Dark|`dark`|yoo-card dark|
|Success|`success`|yoo-card success|
|Danger|`danger`|yoo-card danger|
|Info|`info`|yoo-card info|
|Warning|`warning`|yoo-card warning|
|Gradient Accent|`gradient-accent`|yoo-card gradient-accent|
|Gradient Dark|`gradient-dark`|yoo-card gradient-dark|
|Gradient Success|`gradient-success`|yoo-card gradient-success|
|Gradient Danger|`gradient-danger`|yoo-card gradient-danger|
|Gradient Info|`gradient-info`|yoo-card gradient-info|
|Gradient Warning|`gradient-warning`|yoo-card gradient-warning|

### Style Variation

```yoo-card-style.html
    <yoo-card class="hoverable" sticker="Yoo">
        <div slot="content-slot">I am the slotted content</div>
    </yoo-card>
    <br>
    <br>
    <yoo-card class="accent no-status" sticker="Yoo" is-activable="true">
        <div slot="content-slot">I am the slotted content</div>
    </yoo-card>
    <br>
    <br>
    <yoo-card class="accent hoverable" sticker="Yoo" is-activable="true">
        <div slot="content-slot">I am the slotted content</div>
    </yoo-card>
    <br/>
    <br/>
    <yoo-card class="success hoverable" sticker="Yoo" is-activable="true">
        <div slot="content-slot">I am the slotted content</div>
    </yoo-card>
    <br/>
    <br/>
    <yoo-card class="danger hoverable" sticker="Yoo" is-activable="true">
        <div slot="content-slot">I am the slotted content</div>
    </yoo-card>
    <br/>
    <br/>
    <yoo-card class="info hoverable" sticker="Yoo" is-activable="true">
        <div slot="content-slot">I am the slotted content</div>
    </yoo-card>
    <br/>
    <br/>
    <yoo-card class="warning hoverable" sticker="Yoo" is-activable="true">
        <div slot="content-slot">I am the slotted content</div>
    </yoo-card>
    <br/>
    <br/>
    <yoo-card class="dark hoverable" sticker="Yoo" is-activable="true">
        <div slot="content-slot">I am the slotted content</div>
    </yoo-card>
    <br/>
    <br/>
    <yoo-card class="gradient-accent hoverable" sticker="Yoo" is-activable="true">
        <div slot="content-slot">I am the slotted content</div>
    </yoo-card>
    <br/>
    <br/>
    <yoo-card class="gradient-success hoverable" sticker="Yoo" is-activable="true">
        <div slot="content-slot">I am the slotted content</div>
    </yoo-card>
    <br/>
    <br/>
    <yoo-card class="gradient-danger hoverable" sticker="Yoo" is-activable="true">
        <div slot="content-slot">I am the slotted content</div>
    </yoo-card>
    <br/>
    <br/>
    <yoo-card class="gradient-info hoverable" sticker="Yoo" is-activable="true">
        <div slot="content-slot">I am the slotted content</div>
    </yoo-card>
    <br/>
    <br/>
    <yoo-card class="gradient-warning hoverable" sticker="Yoo" is-activable="true">
        <div slot="content-slot">I am the slotted content</div>
    </yoo-card>
    <br/>
    <br/>
    <yoo-card class="gradient-dark hoverable" sticker="Yoo" is-activable="true">
        <div slot="content-slot">I am the slotted content</div>
    </yoo-card>
```

```yoo-card-style.js
    var comps = document.querySelectorAll('yoo-card');
    var i;
    for(i=0;i<comps.length;i++){
        comps[i].avatarImg = ["./assets/grid/empty-state-1.svg", "./assets/grid/empty-state-1.svg", "./assets/grid/empty-state-1.svg"];
    }
```

## Events
|Attr|Description|
|---|---|---|
|`yooXXXX`|small label|