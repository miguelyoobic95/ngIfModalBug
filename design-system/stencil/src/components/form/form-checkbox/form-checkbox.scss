
@mixin checkbox-color($class, $color) {
    :host(.#{$class}) {
        .container {
            .icon-container {
                 &.enabled {
                    background-color: cssvar($color);
                    border-color: cssvar($color);
                }
            }
        }
    }
}

:host {
    .container {
        display: inline-flex;
        align-items: center;
        .icon-container {
            transition: all 0.3s;
            border-radius: 5px;
            min-width: 1.25rem;
            height: 1.25rem;
            border: 0.1rem solid;
            justify-content: center;
            align-items: center;
            &.enabled {
                background-color: cssvar(dark-40);
                border-color: cssvar(dark-40);
                    &:hover{
                        cursor: pointer;
                        }
                }
            &.disabled {
                background-color: cssvar(dark-20);
                border-color: cssvar(dark-20);
            }
            &.empty {
                background-color: cssvar(light);
                &.enabled {
                    background-color: cssvar(light);
                    &:hover{
                        cursor: pointer;
                        }
                }
                &.disabled {
                    background-color: cssvar(dark-20);
                    border-color: cssvar(dark-20);
                }
            }
                .icon {
                    color: cssvar(light);
                    transition: all 0.3s;
                    font-size: 0.875rem;
                    padding-top: 0.125rem;
                    &.disabled {
                        color: cssvar(dark-80);
                    }
                    &.empty {
                        opacity: 0;
                    }
                }
        }    
        .text-container {
            padding-left: 0.5rem;
            display: flex;
            justify-content: flex-start;
            color: cssvar(dark);
            &.disabled {
                color: cssvar(dark-20);
            }
            &.enabled {
                 &:hover{
                cursor: pointer;
            }
            }   
        }
    }
}

:host(.round) {
    .container {
        .icon-container {
            border-radius: 50%;
        }
    }
}

:host(.large) {
    .container {
        .icon-container {
            width: 2rem;
            height: 2rem;
            .icon {
                font-size: 1rem;
                padding-top: 0rem;
            }
        }
    }
}

@include checkbox-color(accent, accent);
@include checkbox-color(danger, danger);
@include checkbox-color(success, success);
@include checkbox-color(info, info);
@include checkbox-color(warning, warning);
