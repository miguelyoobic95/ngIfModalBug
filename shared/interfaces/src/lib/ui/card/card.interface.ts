import { Â IActionSheetButton } from '../action-sheet/action-sheet.interface';
import { IBadgeEntry } from '../badge/badge.interface';
import { IUser } from '../../entities/user/user.interface';

export type CardType = 'card-feed' | 'card-list' | 'card-default' | 'card-sticky';

export interface ICardEntry {
    heading?: string;
    subheadings?: string[];
    imgSrc?: string;
    topLeftBadge?: string;
    topRightBadge?: string;
    bottomLeftBadge?: string;
    bottomRightBadge?: string;
    type: CardType;
    icon?: string;
    tags?: string[];
    icons?: Array<{icon: string, value?: string, handler?: () => void}>;
    badges?: Array<IBadgeEntry>;
    description?: string;
    actions?: IActionSheetButton[];
    topLeftIcon?: string;
    topRightIcon?: string;
    bottomLeftIcon?: string;
    bottomRightIcon?: string;
    animationName?: string;
    sharedIn?: string;
    groups?: Array<string>;
    bottomAction?: { name: string, handler?: () => void };
    user?: IUser;
    topActions?: Array<{text: string, handler?: () => void}>;
}
