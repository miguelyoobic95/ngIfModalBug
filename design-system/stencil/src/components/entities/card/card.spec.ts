import { TestWindow } from '@stencil/core/testing';
import { YooCardComponent } from './card';
import { YooFormCheckboxComponent } from '../../form/form-checkbox/form-checkbox';

jest.mock('../../../utils/helpers/helpers');

describe('YooCardComponent', () => {
    let window;
    beforeEach(() => {
        window = new TestWindow();
    });
    it('should build', () => {
        expect(new YooCardComponent()).toBeTruthy();
    });

    describe('Snapshots', () => {
        it('Should have a heading', async () => {
            let element = await window.load({
                components: [YooCardComponent],
                html: '<yoo-card heading="Title 1"></yoo-card>'
              });
              expect(element).toMatchSnapshot();
        });
        it('should have list type', async () => {
            let element = await window.load({
                components: [YooCardComponent],
                html: '<yoo-card title="Title 1" type="list"></yoo-card>'
              });
              expect(element).toMatchSnapshot();
        });
        it('Should have an image source', async () => {
            let element = await window.load({
                components: [YooCardComponent],
                html: '<yoo-card img-src="http://www.twitrcovers.com/wp-content/uploads/2012/10/Star-Wars-l.jpg"></yoo-card>'
              });
              expect(element).toMatchSnapshot();
        });
        it('Should be activable', async () => {
            let element = await window.load({
                components: [YooCardComponent],
                html: '<yoo-card is-activable="true" heading="Title 1"></yoo-card>'
              });
              expect(element).toMatchSnapshot();
        });
        it('Should have a badge', async () => {
            let element = await window.load({
                components: [YooCardComponent],
                html: '<yoo-card heading="Title 1" top-left-badge="tl"></yoo-card>'
              });
              expect(element).toMatchSnapshot();
        });
        it('Should have avatar images', async () => {
            let element = await window.load({
                components: [YooCardComponent],
                html: '<yoo-card heading="Title 1"></yoo-card>'
              });
              element.avatarImg = ['./assets/grid/empty-state-1.svg', './assets/grid/empty-state-1.svg', './assets/grid/empty-state-1.svg'];
              expect(element).toMatchSnapshot();
        });
        it('Should have a menu', async () => {
            let element = await window.load({
                components: [YooCardComponent],
                html: '<yoo-card heading="Title 1" has-menu="true"></yoo-card>'
              });
              expect(element).toMatchSnapshot();
        });
        it('Should have date', async () => {
            let element = await window.load({
                components: [YooCardComponent],
                html: '<yoo-card heading="Title 1" date="25/03/1994"></yoo-card>'
              });
              expect(element).toMatchSnapshot();
        });

        it('Should have tags', async () => {
            let element = await window.load({
                components: [YooCardComponent],
                html: '<yoo-card heading="Title 1"></yoo-card>'
              });
              element.tags = [{text: 'Finished', icon: 'yo-flag'}, {text: 'Fire Account', icon: 'yo-fire'}];
              await window.flush(element);
              expect(element).toMatchSnapshot();
        });
    });

    describe('Rendering', () => {
        it ('Should be active when the checkbox is checked', async () => {
            let element = await window.load({
                components: [YooCardComponent],
                html: '<yoo-card heading="Title 1" is-activable="true"></yoo-card>'
              });
            expect(element.getElementsByTagName('yoo-form-checkbox')).toHaveLength(1);
            expect(element.getElementsByClassName('outer-container')).toHaveLength(1);
        });
        it ('Should render a user card', async () => {
            let element = await window.load({
                components: [YooCardComponent],
                html: '<yoo-card heading="Title 1"></yoo-card>'
              });
              expect(element.getElementsByTagName('yoo-avatar')).toHaveLength(0);
              element.avatarShape = 'circle';
              await window.flush();
              expect(element.getElementsByTagName('yoo-avatar')).toHaveLength(1);

        });
        it ('Should remove the checkbox when not activable', async () => {
            let element = await window.load({
                components: [YooCardComponent],
                html: '<yoo-card heading="Title 1" is-activable="true"></yoo-card>'
              });
              expect(element.getElementsByTagName('yoo-form-checkbox')).toHaveLength(1);
              element.isActivable = false;
              await window.flush();
              expect(element.getElementsByTagName('yoo-form-checkbox')).toHaveLength(0);

        });
        it ('Should remove the menu when no-menu is specified', async () => {
            let element = await window.load({
                components: [YooCardComponent],
                html: '<yoo-card has-menu="true" heading="Title 1"></yoo-card>'
              });
              expect(element.getElementsByClassName('yo-more-v')).toHaveLength(1);
              element.hasMenu = false;
              await window.flush();
              expect(element.getElementsByClassName('yo-more-v')).toHaveLength(0);
        });
    });

    describe('Events', () => {
        it('Should emit actionPress event when action is clicked', async () => {
            let element = await window.load({
                components: [YooCardComponent],
                html: '<yoo-card action-button-title="Action"></yoo-card>'
            });
            await window.flush();
            let res;
            let emitItem = jest.fn(ev => {res = ev.detail; });
            window.document.addEventListener('actionPress', emitItem);
            await element.querySelector('div.action-button-container > yoo-button').click();
            expect(emitItem).toHaveBeenCalled();
            expect(res).toEqual(true);
        });

        it('Should emit active event when the checkbox is clicked, should also update active class', async () => {
            let element = await window.load({
                components: [YooCardComponent, YooFormCheckboxComponent],
                html: '<yoo-card is-activable="true"></yoo-card>'
            });
            await window.flush();
            let res;
            let emitItem = jest.fn(ev => {res = ev.detail; });
            window.document.addEventListener('active', emitItem);

            let outerContainer = element.querySelector('.outer-container');
            expect(outerContainer.classList).not.toContain('active');

            let checkbox = element.querySelector('yoo-form-checkbox .text-container');
            await checkbox.click();
            expect(emitItem).toHaveBeenCalled();
            expect(res).toEqual(true);
            await window.flush();
            expect(outerContainer.classList).toContain('active');
            await checkbox.click();
            expect(emitItem).toHaveBeenCalled();
            expect(res).toEqual(false);
            await window.flush();
            expect(outerContainer.classList).not.toContain('active');
        });
    });
});
