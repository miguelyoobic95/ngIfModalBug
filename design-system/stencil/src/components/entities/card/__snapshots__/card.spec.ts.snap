// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooCardComponent Snapshots Should be activable 1`] = `
<yoo-card
  class="hydrated"
  data-yoo-card-host=""
  heading="Title 1"
  is-activable="true"
>
  
  <div
    class="outer-container"
    data-yoo-card=""
  >
    <div
      attr-layout="columns"
      class="image-container"
      data-yoo-card=""
    >
      
      <div
        class="image"
        data-yoo-card=""
        style="background-repeat: no-repeat; background-size: cover; background-position-x: 5;0%; background-image: url(https://cloudinary.com/console/media_library/asset/image/upload/srgyz3etgxqv9vjsz6lv);"
      />
      <yoo-form-checkbox
        class=""
        data-yoo-card=""
      />
    </div>
    <div
      class="status-container"
      data-yoo-card=""
    />
    <div
      class="content-container"
      data-yoo-card=""
    >
      <div
        attr-layout="row"
        class="heading-container"
        data-yoo-card=""
      >
        <span
          class="card-heading"
          data-yoo-card=""
        >
          Title 1
        </span>
        
      </div>
      
      <div
        attr-layout="row"
        class="slot-container"
        data-yoo-card=""
        data-yoo-card-slot=""
      >
        
      </div>
      <div
        attr-layout="row"
        class="avatar-container"
        data-yoo-card=""
      />
      
    </div>
  </div>
</yoo-card>
`;

exports[`YooCardComponent Snapshots Should have a badge 1`] = `
<yoo-card
  class="hydrated"
  data-yoo-card-host=""
  heading="Title 1"
  top-left-badge="tl"
>
  
  <div
    class="outer-container"
    data-yoo-card=""
  >
    <div
      attr-layout="columns"
      class="image-container"
      data-yoo-card=""
    >
      <div
        class="top-left"
        data-yoo-card=""
      >
        <yoo-badge
          class=""
          data-yoo-card=""
          text="tl"
        />
      </div>
      
      <div
        class="image"
        data-yoo-card=""
        style="background-repeat: no-repeat; background-size: cover; background-position-x: 5;0%; background-image: url(https://cloudinary.com/console/media_library/asset/image/upload/srgyz3etgxqv9vjsz6lv);"
      />
      
    </div>
    <div
      class="status-container"
      data-yoo-card=""
    />
    <div
      class="content-container"
      data-yoo-card=""
    >
      <div
        attr-layout="row"
        class="heading-container"
        data-yoo-card=""
      >
        <span
          class="card-heading"
          data-yoo-card=""
        >
          Title 1
        </span>
        
      </div>
      
      <div
        attr-layout="row"
        class="slot-container"
        data-yoo-card=""
        data-yoo-card-slot=""
      >
        
      </div>
      <div
        attr-layout="row"
        class="avatar-container"
        data-yoo-card=""
      />
      
    </div>
  </div>
</yoo-card>
`;

exports[`YooCardComponent Snapshots Should have a heading 1`] = `
<yoo-card
  class="hydrated"
  data-yoo-card-host=""
  heading="Title 1"
>
  
  <div
    class="outer-container"
    data-yoo-card=""
  >
    <div
      attr-layout="columns"
      class="image-container"
      data-yoo-card=""
    >
      
      <div
        class="image"
        data-yoo-card=""
        style="background-repeat: no-repeat; background-size: cover; background-position-x: 5;0%; background-image: url(https://cloudinary.com/console/media_library/asset/image/upload/srgyz3etgxqv9vjsz6lv);"
      />
      
    </div>
    <div
      class="status-container"
      data-yoo-card=""
    />
    <div
      class="content-container"
      data-yoo-card=""
    >
      <div
        attr-layout="row"
        class="heading-container"
        data-yoo-card=""
      >
        <span
          class="card-heading"
          data-yoo-card=""
        >
          Title 1
        </span>
        
      </div>
      
      <div
        attr-layout="row"
        class="slot-container"
        data-yoo-card=""
        data-yoo-card-slot=""
      >
        
      </div>
      <div
        attr-layout="row"
        class="avatar-container"
        data-yoo-card=""
      />
      
    </div>
  </div>
</yoo-card>
`;

exports[`YooCardComponent Snapshots Should have a menu 1`] = `
<yoo-card
  class="hydrated"
  data-yoo-card-host=""
  has-menu="true"
  heading="Title 1"
>
  
  <div
    class="outer-container"
    data-yoo-card=""
  >
    <div
      attr-layout="columns"
      class="image-container"
      data-yoo-card=""
    >
      
      <div
        class="image"
        data-yoo-card=""
        style="background-repeat: no-repeat; background-size: cover; background-position-x: 5;0%; background-image: url(https://cloudinary.com/console/media_library/asset/image/upload/srgyz3etgxqv9vjsz6lv);"
      />
      
    </div>
    <div
      class="status-container"
      data-yoo-card=""
    />
    <div
      class="content-container"
      data-yoo-card=""
    >
      <div
        attr-layout="row"
        class="heading-container"
        data-yoo-card=""
      >
        <span
          class="card-heading"
          data-yoo-card=""
        >
          Title 1
        </span>
        
        <yoo-context-menu
          data-yoo-card=""
        >
          <div
            data-yoo-card=""
            slot="trigger"
          >
            <span
              class="menu-icon"
              data-yoo-card=""
            >
              <i
                class="yo-more-v"
                data-yoo-card=""
              />
            </span>
          </div>
          <div
            class="context-container"
            data-yoo-card=""
            data-yoo-card-slot=""
          >
            
          </div>
        </yoo-context-menu>
      </div>
      
      <div
        attr-layout="row"
        class="slot-container"
        data-yoo-card=""
        data-yoo-card-slot=""
      >
        
      </div>
      <div
        attr-layout="row"
        class="avatar-container"
        data-yoo-card=""
      />
      
    </div>
  </div>
</yoo-card>
`;

exports[`YooCardComponent Snapshots Should have an image source 1`] = `
<yoo-card
  class="hydrated"
  data-yoo-card-host=""
  img-src="http://www.twitrcovers.com/wp-content/uploads/2012/10/Star-Wars-l.jpg"
>
  
  <div
    class="outer-container"
    data-yoo-card=""
  >
    <div
      attr-layout="columns"
      class="image-container"
      data-yoo-card=""
    >
      
      <div
        class="image"
        data-yoo-card=""
        style="background-repeat: no-repeat; background-size: cover; background-position-x: 5;0%; background-image: url(https://cloudinary.com/console/media_library/asset/image/upload/srgyz3etgxqv9vjsz6lv);"
      />
      
    </div>
    <div
      class="status-container"
      data-yoo-card=""
    />
    <div
      class="content-container"
      data-yoo-card=""
    >
      <div
        attr-layout="row"
        class="heading-container"
        data-yoo-card=""
      >
        
      </div>
      
      <div
        attr-layout="row"
        class="slot-container"
        data-yoo-card=""
        data-yoo-card-slot=""
      >
        
      </div>
      <div
        attr-layout="row"
        class="avatar-container"
        data-yoo-card=""
      />
      
    </div>
  </div>
</yoo-card>
`;

exports[`YooCardComponent Snapshots Should have avatar images 1`] = `
<yoo-card
  class="hydrated"
  data-yoo-card-host=""
  heading="Title 1"
>
  
  <div
    class="outer-container"
    data-yoo-card=""
  >
    <div
      attr-layout="columns"
      class="image-container"
      data-yoo-card=""
    >
      
      <div
        class="image"
        data-yoo-card=""
        style="background-repeat: no-repeat; background-size: cover; background-position-x: 5;0%; background-image: url(https://cloudinary.com/console/media_library/asset/image/upload/srgyz3etgxqv9vjsz6lv);"
      />
      
    </div>
    <div
      class="status-container"
      data-yoo-card=""
    />
    <div
      class="content-container"
      data-yoo-card=""
    >
      <div
        attr-layout="row"
        class="heading-container"
        data-yoo-card=""
      >
        <span
          class="card-heading"
          data-yoo-card=""
        >
          Title 1
        </span>
        
      </div>
      
      <div
        attr-layout="row"
        class="slot-container"
        data-yoo-card=""
        data-yoo-card-slot=""
      >
        
      </div>
      <div
        attr-layout="row"
        class="avatar-container"
        data-yoo-card=""
      />
      
    </div>
  </div>
</yoo-card>
`;

exports[`YooCardComponent Snapshots Should have date 1`] = `
<yoo-card
  class="hydrated"
  data-yoo-card-host=""
  date="25/03/1994"
  heading="Title 1"
>
  
  <div
    class="outer-container"
    data-yoo-card=""
  >
    <div
      attr-layout="columns"
      class="image-container"
      data-yoo-card=""
    >
      
      <div
        class="image"
        data-yoo-card=""
        style="background-repeat: no-repeat; background-size: cover; background-position-x: 5;0%; background-image: url(https://cloudinary.com/console/media_library/asset/image/upload/srgyz3etgxqv9vjsz6lv);"
      />
      
    </div>
    <div
      class="status-container"
      data-yoo-card=""
    />
    <div
      class="content-container"
      data-yoo-card=""
    >
      <div
        attr-layout="row"
        class="heading-container"
        data-yoo-card=""
      >
        <span
          class="card-heading"
          data-yoo-card=""
        >
          Title 1
        </span>
        <span
          class="date-card"
          data-yoo-card=""
        >
          25/03/1994
        </span>
        
      </div>
      
      <div
        attr-layout="row"
        class="slot-container"
        data-yoo-card=""
        data-yoo-card-slot=""
      >
        
      </div>
      <div
        attr-layout="row"
        class="avatar-container"
        data-yoo-card=""
      />
      
    </div>
  </div>
</yoo-card>
`;

exports[`YooCardComponent Snapshots Should have tags 1`] = `
<yoo-card
  class="hydrated"
  data-yoo-card-host=""
  heading="Title 1"
>
  
  <div
    class="outer-container"
    data-yoo-card=""
  >
    <div
      attr-layout="columns"
      class="image-container"
      data-yoo-card=""
    >
      
      <div
        class="image"
        data-yoo-card=""
        style="background-repeat: no-repeat; background-size: cover; background-position-x: 5;0%; background-image: url(https://cloudinary.com/console/media_library/asset/image/upload/srgyz3etgxqv9vjsz6lv);"
      />
      
    </div>
    <div
      class="status-container"
      data-yoo-card=""
    />
    <div
      class="content-container"
      data-yoo-card=""
    >
      <div
        attr-layout="row"
        class="heading-container"
        data-yoo-card=""
      >
        <span
          class="card-heading"
          data-yoo-card=""
        >
          Title 1
        </span>
        
      </div>
      
      <div
        attr-layout="row"
        class="slot-container"
        data-yoo-card=""
        data-yoo-card-slot=""
      >
        
      </div>
      <div
        attr-layout="row"
        class="avatar-container"
        data-yoo-card=""
      />
      
    </div>
  </div>
</yoo-card>
`;

exports[`YooCardComponent Snapshots should have list type 1`] = `
<yoo-card
  class="hydrated"
  data-yoo-card-host=""
  title="Title 1"
  type="list"
>
  
  <div
    class="outer-container"
    data-yoo-card=""
  >
    <div
      attr-layout="columns"
      class="image-container"
      data-yoo-card=""
    >
      
      <div
        class="image"
        data-yoo-card=""
        style="background-repeat: no-repeat; background-size: cover; background-position-x: 5;0%; background-image: url(https://cloudinary.com/console/media_library/asset/image/upload/srgyz3etgxqv9vjsz6lv);"
      />
      
    </div>
    <div
      class="status-container"
      data-yoo-card=""
    />
    <div
      class="content-container"
      data-yoo-card=""
    >
      <div
        attr-layout="row"
        class="heading-container"
        data-yoo-card=""
      >
        
      </div>
      
      <div
        attr-layout="row"
        class="slot-container"
        data-yoo-card=""
        data-yoo-card-slot=""
      >
        
      </div>
      <div
        attr-layout="row"
        class="avatar-container"
        data-yoo-card=""
      />
      
    </div>
  </div>
</yoo-card>
`;
